src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -209,21 +210,16 @@
 
     @Override
     public void visitToken(DetailAST ast) {
-        final String line = getLine(ast.getLineNo() - 1);
+        final int[] line = getLineCodePoints(ast.getLineNo() - 1);
         final int before = ast.getColumnNo() - 1;
-        final int[] codePoints = line.codePoints().toArray();
 
-        if ((before == -1 || CommonUtil.isCodePointWhitespace(codePoints, before))
+        if ((before == -1 || CommonUtil.isCodePointWhitespace(line, before))
                 && !isInEmptyForInitializerOrCondition(ast)) {
-            boolean flag = !allowLineBreaks;
-            // verify all characters before '.' are whitespace
-            for (int i = 0; i <= before - 1; i++) {
-                if (!CommonUtil.isCodePointWhitespace(codePoints, i)) {
-                    flag = true;
-                    break;
-                }
-            }
-            if (flag) {
+            final boolean isViolation = !allowLineBreaks
+                    || before > -1
```

Comment:
` before > -1` 3 lines above it is `before == -1`
I do not see assignments to this variable, so it can be boolean variable. As a side note - may be refactor code to check it only once

Also, please rename `before` to something more meaningful.

Reply:
Refactoring code will involve more conditions to be checked in if block, resulting in hard to understand condition.

Reply:
renamed `before` to `columnNoBeforeToken`